overlay_new.py

```diff
@@ -58,16 +64,21 @@ def data_field_coord(x, y):
             CentrePower(self.width, self.height),
             Message(),
         ]
+        self.das_disconnect_message = DASDisconnectMessage()
 
     def on_connect(self, client, userdata, flags, rc):
         print("Connected with rc: {}".format(rc))
 
+    def _display_base_layer(self):
         for component in self.components:
             component.draw_base(self.base_canvas)
 
     def _update_data_layer(self):
         self.data_canvas.clear()
 
+        if not self.client.is_connected:
+            self.das_disconnect_message.draw_data(self.data_canvas)
```

Comment:
Lets move this logic to inside `draw_data` so we don't have to give this class special treatment. Pass a reference to `self.client` into the class through it's constructor, and then whenever we call `draw_data` on the class, we'll be able to check `client.is_connected()` to see if anything should be actually drawn from within the class.

This will mean we can put the `DasDisconnectMessage` in `self.components` with the rest of the `Components` and draw it in the for loop below, super clean!

Reply:
In the current system, components are either drawn only once, or every frame, so unfortunately this wouldn't be possible (again, in the current implementation).

Reply:
> If I'm not wrong, can't we just call `self.das_disconnect_message.draw_data(self.data_canvas)` in the `self.on_disconnect` method fo the parent? It makes more sense calling it there rather than have this `self.client.is_connected` attribute to track it.

I just tried this now, but the problem is that `self.on_disconnect()` only runs when the MQTT was initially running and then it disconnected. So if the broker was disconnected to begin with, then this disconnect message would not display. But you are right when you say it definitely makes sense.

I'm leaning more towards Chris' solution since this would mean that this class can be run inside the for loop of `update_data_layer()` and `draw_base_layer()`, and therefore more maintanable. 

I'll try changing up both approaches and hopefully we come to a solution that we are all satisfied with.